### PR TITLE
Add type annotation to `jnp.tensordot(..., axes=...)`

### PR DESCRIPTION
**Pull request by patrick-kidger**
Just stopping pyright from complaining at me.

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -3156,7 +3156,8 @@ def vdot(a, b, *, precision=None):
 
 
 @util._wraps(np.tensordot, lax_description=_PRECISION_DOC)
-def tensordot(a, b, axes=2, *, precision=None):
+def tensordot(a, b, axes: Union[int, Sequence[int], Sequence[Sequence[int]]] = 2,
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:3159]: As long as we're here, could we annotate `a` and `b` with `ArrayLike` and the return type as `-> Array`?

**Review by jakevdp — CHANGES_REQUESTED** [review]

**Review by jakevdp — APPROVED** [review]

**Comment by patrick-kidger** [comment]
Not sure what's going on with the failing checks, but I think they look unrelated to my change.

**Comment by jakevdp** [comment]
It's related to your change, but cut out of the log:
```
jax/_src/numpy/lax_numpy.py:3172: error: Argument 1 to "len" has incompatible type "int | Sequence[int] | Sequence[Sequence[int]]"; expected "Sized"  [arg-type]
jax/_src/numpy/lax_numpy.py:3173: error: "int" object is not iterable  [misc]
jax/_src/numpy/lax_numpy.py:3178: error: Argument 1 to "len" has incompatible type "Any | int | Sequence[int]"; expected "Sized"  [arg-type]
jax/_src/numpy/lax_numpy.py:3181: error: Item "int" of "Any | int | Sequence[int]" has no attribute "__iter__" (not iterable)  [union-attr]
jax/_src/numpy/lax_numpy.py:3182: error: Item "int" of "Any | int | Sequence[int]" has no attribute "__iter__" (not iterable)  [union-attr]
jax/experimental/shard_map.py:1019: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Found 5 errors in 1 file (checked 1 source file)
```

**Comment by jakevdp** [comment]
#16767 should make the mypy error message in the github CI more informative.

**Comment by patrick-kidger** [comment]
Ah gotcha. Okay, tweaked and let's try again.

**Comment by jakevdp** [comment]
Note that you can run the `mypy` checks locally by doing `git add` on your changes, and then running
```
pre-commit run mypy
```
That would let you iterate more quickly than pushing the changes to this branch and waiting for me to click Approve.

**Pull request merged**